src/main/java/network/brightspots/rcv/CastVoteRecord.java

```diff
@@ -115,10 +134,10 @@ void logRoundOutcome(
 
     StringBuilder logStringBuilder = new StringBuilder();
     logStringBuilder.append("[Round] ").append(round).append(" [CVR] ");
-    if (!isNullOrBlank(suppliedId)) {
-      logStringBuilder.append(suppliedId);
-    } else {
+    if (!isNullOrBlank(computedId)) {
       logStringBuilder.append(computedId);
+    } else {
+      logStringBuilder.append(suppliedId);
     }
```

Comment:
should we show both if they're both present? isn't it plausible that someone reviewing the audit log would want to search by the supplied ID?

Reply:
I wonder if non-unique `suppliedId` values should be a halting exception, or at least a warning. When they're not, presumably either we're not ingesting those IDs properly (e.g. the case you mentioned in which you need to concatenate multiple fields but we were only taking one of those fields) or there's a problem with the source data?

Reply:
This example is from Dominion. In their CVR they really have three separate "ids": TabulatorId, BatchId, and RecordId. Though that data resides in three distinct places in the CVR file, I would bet that Dominion assumes that only the three together can uniquely identify a record.

Our code chose to put RecordId into CastVoteRecord.suppliedId, probably because it looks the most unique: it increments with every record until a new batch or tabulator Id show up (in the data I looked at, often thousands of records). Maybe we could have just left `suppliedId` `null`? Any of the three could make the case that they should be the `suppliedId`. 

I think of the two as 
`suppliedId` = The provider's attempt at a unique identifier, not reliably unique but _can_ be. Helpful sometimes even if it's not unique when navigating vendor files or in their UI and this value is sometimes more visible (despite the data necessary for a unique identifier also potentially being there).
`computedId` = Our attempt at a unique identifier, always reliably unique. Can be used during troubleshooting to identify a single record for troubleshooting.

In your example of us not ingesting properly or there is a problem with the source data it is the non-unique `computedId` that should cause a halting exception. It wouldn't be a surprise when `suppliedId` is not unique.